web/views.py

```
@@ -2730,7 +2743,131 @@ def challenge_submit(request, week_number):
+
+        # Retrieve all submissions for this quiz, ordered by highest score
+        submissions = QuizSubmission.objects.filter(quiz=quiz).order_by("-score", "submitted_at")
+
+        # Get quiz_results from session if present
+        quiz_results = request.session.get('quiz_results', None)
+        if quiz_results:
+            # Clear from session after retrieving
+            del request.session['quiz_results']
+            request.session.modified = True
+
+        end_datetime = timezone.make_aware(datetime.combine(quiz.end_date, quiz.end_time))
```

Ruff (0.8.2)

Undefined name `datetime`

(F821)